streams/src/test/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals.suppress;
+
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Suppress;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.Change;
+import org.apache.kafka.streams.kstream.internals.TimeWindow;
+import org.apache.kafka.streams.processor.MockProcessorContext;
+import org.apache.kafka.test.MockInternalProcessorContext;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.Test;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Random;
+
+import static java.time.Duration.ZERO;
+import static org.apache.kafka.streams.kstream.Suppress.BufferConfig.withBufferFullStrategy;
+import static org.apache.kafka.streams.kstream.Suppress.BufferFullStrategy.SHUT_DOWN;
+import static org.apache.kafka.streams.kstream.Suppress.IntermediateSuppression.withEmitAfter;
+import static org.apache.kafka.streams.kstream.Suppress.emitFinalResultsOnly;
+import static org.apache.kafka.streams.kstream.Suppress.intermediateEvents;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+
+@SuppressWarnings("PointlessArithmeticExpression")
+public class KTableSuppressProcessorTest {
+    private static final Random RANDOM;
+
+    static {
+        final long seed = new Random().nextLong();
+        System.out.println(KTableSuppressProcessorTest.class + ": seed: [" + seed + "]");
+        RANDOM = new Random(seed);
```

Comment:
Decided to start using fuzzing to avoid magic numbers in the tests. Let me know if you prefer it different.